CMakeLists.txt

```diff
@@ -4,13 +4,22 @@ project(marlin VERSION 0.0.1 LANGUAGES CXX)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
-cmake_policy(SET CMP0069 NEW)
-set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
-set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+	cmake_policy(SET CMP0069 NEW)
+	set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
+	set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
 endif()
 
 add_subdirectory(vendor EXCLUDE_FROM_ALL)
 
+if(NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	add_compile_options(--coverage)
```

Comment:
can create coverage directory in build directory using : file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/coverage/"). Otherwise getting error about FileNotFoundError coverage/index.html

Reply:
Fixed, thanks.